### PR TITLE
fix(ui5-combobox): Improve arrow navigation functionality

### PR DESCRIPTION
**Pull request by ndeshev**
Issue: #3790

Note: Also fixes the bugs where the focus of the previously selected item remain duplicated after reopening of the popover and selecting a new one.

### CODE REVIEW

**Review by niyap — CHANGES_REQUESTED** [review]
Now, if you open the suggestions dropdown of aa combobox, navigate through the items and then type something in the input field without closing the dropdown, there is no visual focus over the input field on input. The issue is reproducible, no matter whether the typed value matches suggestions and the dropdown remains open or the value is invalid and it is closed.

File: packages/main/test/specs/ComboBox.spec.js [hunk 1]
```diff
@@ -505,3 +505,110 @@ describe("Accessibility", () => {
...
+describe("Keyboard navigation", () => {
+	it ("Should focus the first item on arrow down and then the input on arrow up", () => {
+		browser.url(`http://localhost:${PORT}/test-resources/pages/ComboBox.html`);
+
+		const combo = $("#combo-grouping");
+		const input = combo.shadow$("#ui5-combobox-input");
+		const arrow = combo.shadow$("[input-icon]");
+		const staticAreaItemClassName = browser.getStaticAreaItemClassName("#combo-grouping");
+		const popover = browser.$(`.${staticAreaItemClassName}`).shadow$("ui5-responsive-popover");
+		let groupItem, listItem;
```
Comment (niyap) [inline packages/main/test/specs/ComboBox.spec.js:518]: Here, it is good to test the value as well. When you are over a group header and when you are over a suggestion

**Review by niyap — CHANGES_REQUESTED** [review]

**Review by MapTo0 — CHANGES_REQUESTED** [review]
When I initially open a combobox via alt + down, the focus should be on the first **selectable** item and it should be autocompleted.

if I arrow down a few times (and a value is autocompleted) then hit backspace the visible focus is not on the list, neither on the combobox

**Exception throw during interaction**

1. Go to "Items with grouping and value state"
2. Open the picker (mouse or alt + down)
3. Go to Argentina with arrows and hit Enter to select it
4. **Open the picker again** (it is closed due to selection) and hit backspace to delete the value
5. Arrow up -> exception is thrown

`Uncaught (in promise) TypeError: Cannot set properties of undefined (setting 'focused')`

File: packages/main/test/specs/ComboBox.spec.js (hunk 1 above)
Comment (ndeshev) [reply packages/main/test/specs/ComboBox.spec.js:518]: I added tests for that previously - when implementing the grouping functionality - the tests are in the "Grouping" module.

**Comment by ndeshev** [comment]
> When I initially open a combobox via alt + down, the focus should be on the first **selectable** item and it should be autocompleted.

You are right, but it is not in the scope of the issue - it's for the basic arrow up/down functionality. 

The other described issues should be fine now.

**Review by niyap — CHANGES_REQUESTED** [review]
Open the test page and check the first sample for example. When focus is on the input field, try to change the selected item without opening the picker, just using arrow up/down keys.
Observed behaviour: visual focus is lost. You have to press Enter in order to return it back to the input field

**Review by niyap — CHANGES_REQUESTED** [review]
When suggestions picker is open and suggestions are too many, so scrollbar is presented, try to navigate between items with arrow down -> when you reach an item that is not visible initially before scrolling, the list is not scrolled and you could not see, which is the selected item now in the list, you can see it, just in the input field which is updated accordingly.

**Comment by MapTo0** [comment]
focus the gruping example, arrow down 1-2 times, then arrow up a few times -> exception is thrown

**Comment by ndeshev** [comment]
I fixed the comments, except for the scrolling suggestions popover problemfor which I created a new issue #3994.

**Review by niyap — CHANGES_REQUESTED** [review]
If the picker is open via F4 or Alt+Arrow Down and then closed via F4 or Alt+ Arrow Up, the visual focus is not presented in the input field

**Review by MapTo0 — APPROVED** [review]
lgtm

**Review by niyap — CHANGES_REQUESTED** [review]
Check the "**Items with grouping and value state**" sample in the test page. Type "A" and navigate with ARROW DOWN through the suggestions, then with Arrow UP go to the value state header and type "r" -> double visual focus is presented, both over the value state header and in the input field. The visual focus should be only in the input field

**Review by niyap — APPROVED** [review]

**Pull request merged**